File: iteration-3-ground-dairy-toothpaste-main/client/src/app/requests/request.ts
export interface Request {
  _id: string;
  itemType: ItemType;
  description: string;
  foodType: FoodType;
  dateAdded: string;
  generalNeed: boolean;}

export type ItemType = 'food' | 'toiletries' | 'other';
export type FoodType = '' | 'dairy' | 'grain' | 'meat' | 'fruit' | 'vegetable';

File: iteration-3-ground-dairy-toothpaste-main/server/src/main/java/umm3601/request/Request.java
package umm3601.request;

import org.mongojack.Id;
import org.mongojack.ObjectId;

@SuppressWarnings({"VisibilityModifier"})

public class Request {
  @ObjectId @Id

  @SuppressWarnings({"MemberName"})

  public String _id;

  public String itemType;
  public String description;
  public String foodType;
  public String dateAdded;
  public Boolean generalNeed;

  @Override
  public boolean equals(Object obj) {
    if (!(obj instanceof Request)) {
      return false;
    }
    Request other = (Request) obj;
    return _id.equals((other._id));
  }

  @Override
  public int hashCode() {
    return _id.hashCode();
  }

}
